logger fixes

vin0110 opened the pull request.

leapinglemming committed:
logger fixes

oxhead commented:
Many debug messages output to the console, which is not desirable.  For example,

```shell
haas config list
```

outputs

```
haas dir: {} /Users/oxhead/.haas
haas config dir: {} /Users/oxhead/.haas/config
Loading config from {} /Users/oxhead/.haas/config/mycluster1.yaml
{<HaasConfigurationKey.HAAS_WORKER_NUM_NODES: 'haas_worker_num_nodes'>: 1, <HaasConfigurationKey.HAAS_EBS_VOLUME_SIZE: 'haas_ebs_volume_size'>: 20, <HaasConfigurationKey.HAAS_SERVICE_INSTANCE_TYPE: 'haas_service_instance_type'>: 'c4.large', <HaasConfigurationKey.HAAS_WORKER_INSTANCE_TYPE: 'haas_worker_instance_type'>: 'c4.large', <HaasConfigurationKey.HAAS_EC2_KEY_PAIR: 'haas_ec2_key_pair'>: 'osr', <HaasConfigurationKey.HAAS_SERVICE_NUM_NODES: 'haas_service_num_nodes'>: 1}
Loading config from {} /Users/oxhead/.haas/config/mycluster_test.yaml
{<HaasConfigurationKey.HAAS_WORKER_NUM_NODES: 'haas_worker_num_nodes'>: 1, <HaasConfigurationKey.HAAS_EBS_VOLUME_SIZE: 'haas_ebs_volume_size'>: 20, <HaasConfigurationKey.HAAS_SERVICE_INSTANCE_TYPE: 'haas_service_instance_type'>: 't2.micro', <HaasConfigurationKey.HAAS_WORKER_INSTANCE_TYPE: 'haas_worker_instance_type'>: 't2.micro', <HaasConfigurationKey.HAAS_EC2_KEY_PAIR: 'haas_ec2_key_pair'>: 'haas', <HaasConfigurationKey.HAAS_SERVICE_NUM_NODES: 'haas_service_num_nodes'>: 1}
Loading config from {} /Users/oxhead/.haas/config/test1.yaml
{<HaasConfigurationKey.HAAS_WORKER_NUM_NODES: 'haas_worker_num_nodes'>: 1, <HaasConfigurationKey.HAAS_EBS_VOLUME_SIZE: 'haas_ebs_volume_size'>: 20, <HaasConfigurationKey.HAAS_SERVICE_INSTANCE_TYPE: 'haas_service_instance_type'>: 't2.micro', <HaasConfigurationKey.HAAS_WORKER_INSTANCE_TYPE: 'haas_worker_instance_type'>: 't2.micro', <HaasConfigurationKey.HAAS_EC2_KEY_PAIR: 'haas_ec2_key_pair'>: 'haas', <HaasConfigurationKey.HAAS_SERVICE_NUM_NODES: 'haas_service_num_nodes'>: 1}
Loading config from {} /Users/oxhead/.haas/config/test2.yaml
{<HaasConfigurationKey.HAAS_WORKER_NUM_NODES: 'haas_worker_num_nodes'>: 1, <HaasConfigurationKey.HAAS_EBS_VOLUME_SIZE: 'haas_ebs_volume_size'>: 20, <HaasConfigurationKey.HAAS_SERVICE_INSTANCE_TYPE: 'haas_service_instance_type'>: 't2.micro', <HaasConfigurationKey.HAAS_WORKER_INSTANCE_TYPE: 'haas_worker_instance_type'>: 't2.micro', <HaasConfigurationKey.HAAS_EC2_KEY_PAIR: 'haas_ec2_key_pair'>: 'haas', <HaasConfigurationKey.HAAS_SERVICE_NUM_NODES: 'haas_service_num_nodes'>: 1}
Loading config from {} /Users/oxhead/.haas/config/test3.yaml
{<HaasConfigurationKey.HAAS_WORKER_NUM_NODES: 'haas_worker_num_nodes'>: 1, <HaasConfigurationKey.HAAS_EBS_VOLUME_SIZE: 'haas_ebs_volume_size'>: 20, <HaasConfigurationKey.HAAS_SERVICE_INSTANCE_TYPE: 'haas_service_instance_type'>: 't2.micro', <HaasConfigurationKey.HAAS_WORKER_INSTANCE_TYPE: 'haas_worker_instance_type'>: 't2.micro', <HaasConfigurationKey.HAAS_EC2_KEY_PAIR: 'haas_ec2_key_pair'>: 'haas', <HaasConfigurationKey.HAAS_SERVICE_NUM_NODES: 'haas_service_num_nodes'>: 1}
Loading config from {} /Users/oxhead/.haas/config/test4.yaml
{<HaasConfigurationKey.HAAS_WORKER_NUM_NODES: 'haas_worker_num_nodes'>: 1, <HaasConfigurationKey.HAAS_EBS_VOLUME_SIZE: 'haas_ebs_volume_size'>: 20, <HaasConfigurationKey.HAAS_SERVICE_INSTANCE_TYPE: 'haas_service_instance_type'>: 't2.micro', <HaasConfigurationKey.HAAS_WORKER_INSTANCE_TYPE: 'haas_worker_instance_type'>: 't2.micro', <HaasConfigurationKey.HAAS_EC2_KEY_PAIR: 'haas_ec2_key_pair'>: 'haas', <HaasConfigurationKey.HAAS_SERVICE_NUM_NODES: 'haas_service_num_nodes'>: 1}
{'mycluster_test': <haascli.config.HaaSConfiguration object at 0x10fa92e80>, 'test1': <haascli.config.HaaSConfiguration object at 0x110e19ac8>, 'mycluster1': <haascli.config.HaaSConfiguration object at 0x110e19b70>, 'test2': <haascli.config.HaaSConfiguration object at 0x110e19da0>, 'test4': <haascli.config.HaaSConfiguration object at 0x110e19f98>, 'test3': <haascli.config.HaaSConfiguration object at 0x110e19c88>}
```

vin0110 commented:
They are annoying. However, they are from config (specifically, test_config_manager.py). It was printing using message, which is output to the user. It was erroneously re-directed to INFO.

It isn't clear to me whether you wanted those message written to log or console. Also, shouldn't the data be formatted?

oxhead committed:
Few enhancements.

1. Fixed an issue writing to the log file.
2. Integrate coloredlogs.
3. Fixed some syntax errors.

oxhead committed:
Initialize the logging without calling setup_logging.

oxhead committed:
Fixed a typo in the error message.
